2024.6.0

frenck opened the pull request:
## ℹ️ Do not merge!

This is the release PR for Home Assistant 2024.6.0, to be released on _June the 5th 2024_.

~~This PR exists and is here, to ensure our builds are fully passing during our beta week.~~

Beta week is over, this is now turned into the release PR, ready for approval.

tronikos committed:
Add diagnostics to Google Generative AI (#118262)

* Add diagnostics for Google Generative AI

* Remove quality scale from manifest

* include options in diagnostics

bdraco committed:
Migrate mqtt discovery subscribes to use internal helper (#118279)

tronikos committed:
Mock llm prompts in test_default_prompt for Google Generative AI (#118286)

tsvi committed:
Change strings to const in Jewish Calendar (#118274)

mib1185 committed:
Bump aiovlc to 0.3.2 (#118258)

bdraco committed:
Fix flakey bootstrap test (#118285)

Bre77 committed:
Use entry.runtime_data in Tessie (#118287)

scop committed:
Add additional Huawei LTE 5G sensors (#108928)

* Add some Huawei LTE 5G sensor descriptions

Closes https://github.com/home-assistant/core/issues/105786

* Mark cqi1 and nrcqi1 as diagnostic

Commit:
Fix some typing errors in Bring integration (#115641)

Fix typing errors

erwindouna committed:
Add DSMR Reader tests (#115808)

* Add DSMR Reader sensor tests

* Change to paramatization

* Removing patch

* Emulate the test

* Go for 100% test coverage

* Adding defintions.py

* Add myself as code owner to keep improving

bdraco committed:
Migrate mqtt mixin async_added_to_hass inner functions to bound methods (#118280)

ollo69 committed:
Add diagnostics test to AndroidTV (#117129)

lboue committed:
Add Matter fan platform (#111212)

Co-authored-by: Marcel van der Veldt <m.vanderveldt@outlook.com>

starkillerOG committed:
Bump reolink-aio to 0.8.11 (#118294)

marcelveldt committed:
Fix Matter device ID for non-bridged composed device (#118256)

epenet committed:
Use registry fixtures in tests (tailscale) (#118301)

slovdahl committed:
Add reconfigure step for vallox (#115915)

* Add reconfigure step for vallox

* Reuse translation

epenet committed:
Use registry fixtures in tests (a-h) (#118288)

epenet committed:
Use registry fixtures in tests (m-n) (#118291)

epenet committed:
Use registry fixtures in tests (o-p) (#118292)

epenet committed:
Use registry fixtures in scaffold (#118308)

epenet committed:
Use registry fixtures in tests (t-u) (#118297)

epenet committed:
Use registry fixtures in tests (r) (#118293)

epenet committed:
Use registry fixtures in tests (v-y) (#118299)

Poshy163 committed:
Thread: Add more Thread vendor to brand mappings (#115888)

Co-authored-by: Stefan Agner <stefan@agner.ch>
Co-authored-by: Robert Resch <robert@resch.dev>

epenet committed:
Use registry fixtures in tests (s) (#118295)

mdegat01 committed:
Add more supervisor info to system info panel (#115715)

* Add virtualization field fo system info

* Add ntp sync and host connectivity

* Prevent nonetype errors

* Add supervisor_connectivity and fix tests

* Add mock of network info to other fixtures

* Update more fixtures with network/info mock

balloob committed:
Handle generic commands as area commands in the LLM Assist API (#118276)

* Handle generic commands as area commands in the LLM Assist API

* Add word area

chatziko committed:
Don't set 'assist in progess' flag on wake_word-end (#113585)

epenet committed:
Enforce namespace use for import conventions (#118215)

* Enforce namespace use for import conventions

* Include all registries

* Only apply to functions

* Use blacklist

* Rephrase comment

* Add async_entries_for_config_entry

* Typo

* Improve

* More core files

* Revert "More core files"

This reverts commit 9978b9370629af402a9a18f184b6f3a7ad45b08d.

* Revert diagnostics amends

* Include category/floor/label registries

* Performance

* Adjust text

synesthesiam committed:
Bump intents and add tests for new error messages (#118317)

* Add new error keys

* Bump intents and test new error messages

* Fix response text

epenet committed:
Use registry fixtures in tests (z) (#118300)

epenet committed:
Add more type hints to pylint plugin (#118319)

synesthesiam committed:
Add temperature slot to light turn on intent (#118321)

synesthesiam committed:
Respect WyomingSatelliteMuteSwitch state on start (#118320)

* Respect WyomingSatelliteMuteSwitch state on start

* Fix test

---------

Co-authored-by: Kostas Chatzikokolakis <kostas@chatzi.org>

jesserockz committed:
[esphome] Assist timers (#118275)

* [esphome] Assist timers

* Add intent to manifest dependencies

* Add test

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

dontinelli committed:
Raise UpdateFailed on fyta API error (#118318)

* Raise UpdateFailed

* Update homeassistant/components/fyta/coordinator.py

Co-authored-by: Robert Resch <robert@resch.dev>

* Remove logger

* simplify code

---------

Co-authored-by: Robert Resch <robert@resch.dev>

synesthesiam committed:
Add platforms to intent handlers (#118328)

bdraco committed:
Speed up mqtt debug info on message callback (#118303)

elupus committed:
Update ha philips_js to 3.2.2 (#118326)

synesthesiam committed:
Filter timers more when pausing/unpausing (#118331)

bdraco committed:
Mark sonos group update a background task (#118333)

jesserockz committed:
[esphome] 100% voice assistant test coverage (#118334)

ludeeus committed:
Use None default for traccar server battery level sensor (#118324)

Do not set -1 as default for traccar server battery level

marcelveldt committed:
Several fixes for the Matter climate platform (#118322)

* extend hvacmode mapping with extra modes

* Fix climate platform

* adjust tests

* fix reversed test

* cleanup

* dry and fan hvac mode test

mj23000 committed:
Fix source_change not triggering an update (#118312)

balloob committed:
Update the recommended model for Google Gen AI (#118323)

bdraco committed:
Use del instead of pop in the entity platform remove (#118337)

bdraco committed:
Replace pop calls with del where the result is discarded in mqtt (#118338)

bdraco committed:
Replace pop calls with del where the result is discarded in entity (#118340)

balloob committed:
LLM Assist API to ignore intents if not needed for exposed entities or calling device (#118283)

* LLM Assist API to ignore timer intents if device doesn't support it

* Refactor to use API instances

* Extract ToolContext class

* Limit exposed intents based on exposed entities

synesthesiam committed:
Add Conversation command to timers (#118325)

* Add Assist command to timers

* Rename to conversation_command. Execute in timer code.

* Make agent_id optional

* Fix arg

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

balloob committed:
Tweak Assist LLM API prompt (#118343)

bdraco committed:
Small speed up to connecting dispatchers (#118342)

balloob committed:
Reduce the intent response data sent to LLMs (#118346)

* Reduce the intent response data sent to LLMs

* No longer delete speech

bdraco committed:
Add cache to more complex entity filters (#118344)

Many of these do regexes and since the entity_ids are almost
always the same we should cache these

bdraco committed:
Improve websocket message coalescing to handle thundering herds better (#118268)

* Increase websocket peak messages to match max expected entities

During startup the websocket would frequently disconnect if more than
4096 entities were added back to back. Some MQTT setups will have more
than 10000 entities. Match the websocket peak value to the max expected
entities

* coalesce more

* delay more if the backlog gets large

* wait to send if the queue is building rapidly

* tweak

* tweak for chrome since it works great in firefox but chrome cannot handle it

* Revert "tweak for chrome since it works great in firefox but chrome cannot handle it"

This reverts commit 439e2d76b11d2355c552c8a577d0e85fc7262808.

* adjust for chrome

* lower number

* remove code

* fixes

* fast path for bytes

* compact

* adjust test since we see the close right away now on overload

* simplify check

* reduce loop

* tweak

* handle ready right away

bdraco committed:
Replace pop calls with del where the result is discarded in restore_state (#118339)

bdraco committed:
Fix last_reported_timestamp not being updated when last_reported is changed (#118341)

* Reduce number of calls to last_reported_timestamp

When a state is created, last_update is always the same
as last_reported, and we only update it later if it changes
so we can pre-set the cached property to avoid it being
run when the recorder accesses it later.

* fix cache not being overridden

* coverage

jpbede committed:
Use runtime_data in ping (#118332)

Quentame committed:
Fix epic_games_store mystery game URL (#118314)

emontnemery committed:
Don't report entities with invalid unique id when loading the entity registry (#118290)

gjohansson-ST committed:
Add smoke detector temperature to Yale Smart Alarm (#116306)

m10x committed:
Add august open action (#113795)

Co-authored-by: J. Nick Koston <nick@koston.org>

osohotwateriot committed:
Add OSO Energy binary sensors (#117174)

raman325 committed:
Update zwave_js WS APIs for provisioning (#117400)

epenet committed:
Improve typing for `calls` fixture in tests (a-l) (#118349)

* Improve typing for `calls` fixture in tests (a-l)

* More

* More

edenhaus committed:
Add ws endpoint to remove expiration date from refresh tokens (#117546)

Co-authored-by: Erik Montnemery <erik@montnemery.com>

dependabot[bot] committed:
Bump docker/login-action from 3.1.0 to 3.2.0 (#118351)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Bygood91 committed:
Add Google assistant Gate device type (#118144)

bdraco committed:
Make Recorder dialect_name a cached_property (#117922)

epenet committed:
Rename calls fixture in calendar tests (#118353)

epenet committed:
Rename calls fixture in mqtt tests (#118354)

Rename calls fixture in mqtt

epenet committed:
Rename calls fixture in components tests (#118355)

epenet committed:
Improve typing for `calls` fixture in tests (m-z) (#118350)

* Improve typing for `calls` fixture in tests (m-z)

* More

* More

agners committed:
Adjust add-on installation error message (#118309)

epenet committed:
Rename service_calls fixture in template tests (#118358)

gjohansson-ST committed:
Add more tests to Yale Smart Alarm (#116501)

erwindouna committed:
Add DSMR Reader icons (#118329)

gjohansson-ST committed:
Add default code to alarm_control_panel (#112540)

bouwew committed:
Discover new device at runtime in Plugwise (#117688)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

jbouwh committed:
Allow MQTT device based auto discovery (#109030)

* Add MQTT device based auto discovery

* Respect override of component options over shared ones

* Add state_topic, command_topic, qos and encoding as shared options

* Add shared option test

* Rename device.py to schemas.py

* Remove unused legacy `platform` attribute to avoid confusion

* Split validation device and origin info

* Require `origin` info on device based discovery

* Log origin info for only once for device discovery

* Fix tests and linters

* ruff

* speed up _replace_all_abbreviations

* Fix imports and merging errors - add slots attr

* Fix unrelated const changes

* More unrelated changes

* join string

* fix merge

* Undo move

* Adjust logger statement

* fix task storm to load platforms

* Revert "fix task storm to load platforms"

This reverts commit 8f12a5f2511ab872880a186f5a4605c8bae80c7d.

* bail if logging is disabled

* Correct mixup object_id and node_id

* Auto migrate entities to device discovery

* Add device discovery test for device_trigger

* Add migration support for non entity platforms

* Use helper to remove discovery payload

* Fix tests after update branch

* Add discovery migration test

* Refactor

* Repair after rebase

* Fix discovery is broken after migration

* Improve comments

* More comment improvements

* Split long lines

* Add comment to indicate payload dict can be empty

* typo

* Add walrus and update comment

* Add tag to migration test

* Join try blocks

* Refactor

* Cleanup not used attribute

* Refactor

* Move _replace_all_abbreviations out of try block

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

Nezz committed:
Extend image_upload to return the original image (#116652)

ludeeus committed:
Update pylaunches dependency to version 2.0.0 (#118362)

dontinelli committed:
Set quality scale of fyta to platinum (#118307)

tronikos committed:
Fix issue when you have multiple Google Generative AI config entries and you remove one of them (#118365)

karwosts committed:
Support in blueprint schema for input sections (#110513)

* initial commit for sections

* updates

* add description

* fix test

* rename collapsed key

* New schema

* update snapshots

* Testing for sections

* Validate no duplicate input keys across sections

* rename all_inputs

* Update homeassistant/components/blueprint/models.py

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

tronikos committed:
Expose useful media player attributes to LLMs (#118363)

tronikos committed:
Fix LLM tracing for Google Generative AI (#118359)

Fix LLM tracing for Gemini

tronikos committed:
Ask LLM to pass area name and domain (#118357)

tronikos committed:
Default to gemini-1.5-flash-latest in Google Generative AI (#118367)

Default to flash

epenet committed:
Add LogCaptureFixture type hints in tests (#118372)

epenet committed:
Add SnapshotAssertion type hints in tests (#118371)

osohotwateriot committed:
Bump pyosoenergyapi to 1.1.4 (#118368)

epenet committed:
Add AiohttpClientMocker type hints in tests (#118373)

epenet committed:
Add ClientSessionGenerator type hints in tests (#118377)

joostlek committed:
Bump yt-dlp to 2024.05.27 (#118378)

tronikos committed:
Add Android TV Remote debug logs to help with zeroconf issue (#117960)

mib1185 committed:
Rework and simplify the cleanup of orphan AVM Fritz!Tools entities (#117706)

mj23000 committed:
Convert unnecessary coroutines into functions (#118311)

autinerd committed:
Bump ruff to 0.4.6 (#118384)

gjohansson-ST committed:
Entity for Tags (#115048)

Co-authored-by: Robert Resch <robert@resch.dev>
Co-authored-by: Erik <erik@montnemery.com>

joostlek committed:
Bump airgradient to 0.4.2 (#118389)

synesthesiam committed:
Allow delayed commands to not have a device id (#118390)

bramkragten committed:
Bump frontend to 20240529.0 (#118392)

marcelveldt committed:
Bump Python Matter Server library to 6.1.0(b0) (#118388)

TheJulianJES committed:
Bump zha-quirks to 0.0.116 (#118393)

joostlek committed:
Add select entities to AirGradient (#117136)

frenck committed:
Bump version to 2024.6.0b0

joostlek commented:
> This PR exists and is here, to ensure our builds are fully passing during our beta week.

How does that work compared to for example last month?

frenck commented:
> > This PR exists and is here, to ensure our builds are fully passing during our beta week.
> 
> How does that work compared to for example last month?

Nothing is different, it only puts CI error in our face (instead of discovering when we make the release PR hours before release).

marcelveldt committed:
Fix light discovery for Matter dimmable plugin unit (#118404)

marcelveldt committed:
Add translation strings for Matter Fan presets (#118401)

synesthesiam committed:
Handle case where timer device id exists but is not registered (delayed command) (#118410)

Handle case where device id exists but is not registered

bdraco committed:
Fix google_mail doing blocking I/O in the event loop (#118421)

fixes #118411

bdraco committed:
Fix google_tasks doing blocking I/O in the event loop (#118418)

fixes #118407

bdraco committed:
Ensure paho.mqtt.client is imported in the executor (#118412)

fixes #118405

bdraco committed:
Fix workday doing blocking I/O in the event loop (#118422)

swcloudgenie committed:
New official genie garage integration (#117020)

* new official genie garage integration

* move api constants into api module

* move scan interval constant to cover.py

freekode committed:
Fix OpenWeatherMap migration (#118428)

bdraco committed:
Fix blocking I/O in the event loop in meteo_france (#118429)

tronikos committed:
Bump opower to 0.4.6 (#118434)

bdraco committed:
Fix google_mail doing blocking i/o in the event loop (take 2) (#118441)

frenck committed:
Bump version to 2024.6.0b1

tronikos committed:
Improve LLM prompt (#118443)

* Improve LLM prompt

* test

* improvements

* improvements

tronikos committed:
Instruct LLM to not pass a list to the domain (#118451)

bieniu committed:
Raise `ConfigEntryNotReady` when there is no `_id` in the Tractive data (#118467)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

edenhaus committed:
Bump deebot-client to 7.3.0 (#118462)

kibergus committed:
Add Total Volatile Organic Compounds (tVOC) matter discovery schema (#116963)

lunmay committed:
Typo fix in media_extractor (#118473)

edenhaus committed:
Set enity_category to config for airgradient select entities (#118477)

marcelveldt committed:
Mark Matter climate dry/fan mode support on Panasonic AC (#118485)

bramkragten committed:
Update frontend to 20240530.0 (#118489)

marcelveldt committed:
Extend Matter sensor discovery schemas for Air Purifier / Air Quality devices (#118483)

Co-authored-by: Franck Nijhof <git@frenck.dev>

frenck committed:
Bump version to 2024.6.0b2

janiversen reviewed (approved):
Modbus lgtm, thanks.

tsvi committed:
Fix Jewish calendar unique id's (#117985)

* Initial commit

* Fix updating of unique id

* Add testing to check the unique id is being updated correctly

* Reload the config entry and confirm the unique id has not been changed

* Move updating unique_id to __init__.py as suggested

* Change the config_entry variable's name back from config to config_entry

* Move the loop into the update_unique_ids method

* Move test from test_config_flow to test_init

* Try an early optimization to check if we need to update the unique ids

* Mention the correct version

* Implement suggestions

* Ensure all entities are migrated correctly

* Just to be sure keep the previous assertion as well

balloob committed:
Intent script: allow setting description and platforms (#118500)

* Add description to intent_script

* Allow setting platforms

balloob committed:
Ignore the toggle intent (#118491)

gjohansson-ST committed:
Fix blocking call in holiday (#118496)

jbouwh committed:
Fix group platform dependencies (#118499)

dontinelli committed:
Adjustment of unit of measurement for light (#116695)

zxdavb committed:
Fix evohome so it doesn't retrieve schedules unnecessarily (#118478)

gjohansson-ST committed:
Remove not needed hass object from Tag (#118498)

dgomes committed:
Add support for V2C Trydan 2.1.7 (#117147)

* Support for firmware 2.1.7

* add device ID as unique_id

* add device ID as unique_id

* add test device id as unique_id

* backward compatibility

* move outside try

* Sensor return type

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* not needed

* make slave error enum state

* fix enum

* Update homeassistant/components/v2c/sensor.py

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/v2c/strings.json

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update homeassistant/components/v2c/strings.json

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* simplify tests

* fix misspellings from upstream library

* add sensor tests

* just enough coverage for enum sensor

* Refactor V2C tests (#117264)

* Refactor V2C tests

* fix rebase issues

* ruff

* review

* fix https://github.com/home-assistant/core/issues/117296

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

tronikos committed:
Ignore deprecated open and close cover intents for LLMs (#118515)

gjohansson-ST committed:
Fix tado non-string unique id for device trackers (#118505)

* Fix tado none string unique id for device trackers

* Add comment

* Fix comment

gjohansson-ST committed:
Fix key issue in config entry options in Openweathermap (#118506)

tronikos committed:
Fix LLMs asking which area when there is only one device (#118518)

* Ignore deprecated open and close cover intents for LLMs

* Fix LLMs asking which area when there is only one device

* remove unrelated changed

* remove unrelated changes

tronikos committed:
Fix unnecessary single quotes escaping in Google AI (#118522)

tronikos committed:
Improve LLM prompt (#118520)

zweckj committed:
Add typing for OpenAI client and fallout (#118514)

* typing for client and consequences

* Update homeassistant/components/openai_conversation/conversation.py

---------

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

jpbede committed:
Log aiohttp error in rest_command (#118453)

balloob committed:
Bump version to 2024.6.0b3

tsvi committed:
Fix YAML deprecation breaking version in jewish calendar and media extractor (#118546)

* Fix YAML deprecation breaking version

* Update

* fix media extractor deprecation as well

* Add issue_domain

joostlek committed:
Fix typo in OWM strings (#118538)

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

bdraco committed:
Fix snmp doing blocking I/O in the event loop (#118521)

luca-angemi committed:
Fix telegram doing blocking I/O in the event loop (#118531)

bieniu committed:
In Brother integration use SnmpEngine from SNMP integration (#118554)

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>
Co-authored-by: J. Nick Koston <nick@koston.org>

Bre77 committed:
Fix off_grid_vehicle_charging_reserve_percent in Teselemetry (#118532)

basbruss committed:
Revert "Fix Tibber sensors state class" (#118409)

Revert "Fix Tibber sensors state class (#117085)"

This reverts commit 658c1f3d97a8a8eb0d91150e09b36c995a4863c5.

bdraco committed:
Fix openweathermap config entry migration (#118526)

* Fix openweathermap config entry migration

The options keys were accidentally migrated to data so
they could no longer be changed in the options flow

* more fixes

* adjust

* reduce

* fix

* adjust

chishm committed:
Fix KeyError in dlna_dmr SSDP config flow when checking existing config entries (#118549)

Fix KeyError checking existing dlna_dmr config entries

balloob committed:
Rename llm.ToolContext to llm.LLMContext (#118566)

zweckj committed:
Migrate openai_conversation to `entry.runtime_data` (#118535)

* switch to entry.runtime_data

* check for missing config entry

* Update homeassistant/components/openai_conversation/__init__.py

---------

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

balloob committed:
Fix openAI tool calls (#118577)

Shulyaka committed:
Add OpenAI Conversation system prompt `user_name` and `llm_context` variables (#118512)

* OpenAI Conversation: Add variables to the system prompt

* User name and llm_context

* test for user name

* test for user id

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

Shulyaka committed:
Add Google Generative AI Conversation system prompt `user_name` and `llm_context` variables (#118510)

* Google Generative AI Conversation: Add variables to the system prompt

* User name and llm_context

* test for template variables

* test for template variables

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

epenet committed:
Add ability to replace connections in DeviceRegistry (#118555)

* Add ability to replace connections in DeviceRegistry

* Add more tests

* Improve coverage

* Apply suggestion

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

---------

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

balloob committed:
Bump version to 2024.6.0b4

jpbede committed:
Run ruff format for device registry (#118582)

epenet committed:
Update device connections in samsungtv (#118556)

tronikos committed:
Stop instructing LLM to not pass the domain as a list (#118590)

balloob committed:
Add base prompt for LLMs (#118592)

tronikos committed:
Strip Google AI text responses (#118593)

* Strip Google AI test responses

* strip each part

tronikos committed:
Use gemini-1.5-flash-latest in google_generative_ai_conversation.generate_content (#118594)

thytterdal committed:
Ignore myuplink sensors without a description that provide non-numeric values (#115525)

Ignore sensors without a description that provide non-numeric values

Co-authored-by: Jan-Philipp Benecke <jan-philipp@bnck.me>

mib1185 committed:
Avoid future exception during setup of Synology DSM (#118583)

* avoid future exception during integration setup

* clear future flag during setup

* always clear the flag (with comment)

luca-angemi committed:
Fix telegram bot send_document (#118616)

epenet committed:
Address late review comment in samsungtv (#118539)

Address late comment in samsungtv

bdraco committed:
Include a traceback for non-strict event loop blocking detection (#118620)

mrpasztoradam committed:
Bump pyads to 3.4.0 (#116934)

Co-authored-by: J. Nick Koston <nick@koston.org>

joostlek committed:
Fix incorrect placeholder in SharkIQ (#118640)

Update strings.json

jbouwh committed:
Fix handling undecoded mqtt sensor payloads (#118633)

puddly committed:
Bump ZHA dependencies (#118658)

* Bump bellows to 0.39.0

* Do not create a backup if there is no active ZHA gateway object

* Bump universal-silabs-flasher as well

gjohansson-ST committed:
Fix entity state dispatching for Tag entities (#118662)

balloob committed:
Bump version to 2024.6.0b5

rikroe committed:
Set lock state to unkown on BMW API error (#118559)

* Revert to previous lock state on BMW API error

* Set lock state to unkown on error and force refresh from API

---------

Co-authored-by: Richard <rikroe@users.noreply.github.com>

bdraco committed:
Revert "Add websocket API to get list of recorded entities (#92640)" (#118644)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

iMicknl committed:
Bump pyoverkiz to 1.13.11 (#118703)

emontnemery committed:
Don't store tag_id in tag storage (#118707)

rikroe committed:
Refactor fixture calling for BMW tests (#118708)

* Refactor BMW tests to use pytest.mark.usefixtures

* Fix freeze_time

---------

Co-authored-by: Richard <rikroe@users.noreply.github.com>

piitaya committed:
Bump python-roborock to 2.2.2 (#118697)

joostlek committed:
Disable both option in Airgradient select (#118702)

epenet committed:
Bump renault-api to 0.2.3 (#118718)

matrixd2 committed:
Fixing device model compatibility issues. (#118686)

balloob committed:
Automatically fill in slots based on LLM context (#118619)

* Automatically fill in slots from LLM context

* Add tests

* Apply suggestions from code review

Co-authored-by: Allen Porter <allen@thebends.org>

---------

Co-authored-by: Allen Porter <allen@thebends.org>

emontnemery committed:
Tweak light service schema (#118720)

emontnemery committed:
Remove tag_id from tag store (#118713)

jpbede committed:
Rename Discovergy to inexogy (#118724)

mkmer committed:
Remove unintended translation key from blink (#118712)

dgomes committed:
Address reviews comments in #117147 (#118714)

joostlek committed:
Configure device in airgradient config flow (#118699)

bramkragten committed:
Update frontend to 20240603.0 (#118736)

dknowles2 committed:
Bump pydrawise to 2024.6.2 (#118608)

balloob committed:
Remove dispatcher from Tag entity (#118671)

* Remove dispatcher from Tag entity

* type

* Don't use  helper

* Del is faster than pop

* Use id in update

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

rikroe committed:
Use snapshot_platform helper for BMW tests (#118735)

* Use snapshot_platform helper

* Remove comments

---------

Co-authored-by: Richard <rikroe@users.noreply.github.com>

tronikos committed:
Use ISO format when passing date to LLMs (#118705)

synesthesiam committed:
Resolve areas/floors to ids in intent_script (#118734)

joostlek committed:
Require firmware version 3.1.1 for airgradient (#118744)

frenck committed:
Bump version to 2024.6.0b6

zxdavb committed:
Harden evohome against failures to retrieve zone schedules (#118517)

balloob committed:
Update OpenAI prompt on each interaction (#118747)

balloob committed:
Google Gen AI: Copy messages to avoid changing the trace data (#118745)

jbouwh committed:
Revert "Allow MQTT device based auto discovery" (#118746)

Revert "Allow MQTT device based auto discovery (#109030)"

This reverts commit 585892f0678dc054819eb5a0a375077cd9b604b8.

jbouwh committed:
Do not log mqtt origin info if the log level does not allow it (#118752)

synesthesiam committed:
Clean up exposed domains (#118753)

* Remove lock and script

* Add media player

* Fix tests

synesthesiam committed:
Bump intents to 2024.6.3 (#118748)

jbouwh committed:
Recover mqtt abbrevations optimizations (#118762)

Co-authored-by: J. Nick Koston <nick@koston.org>

ludeeus committed:
Update hass-nabucasa to version 0.81.1 (#118768)

starkillerOG committed:
Bump reolink-aio to 0.9.1 (#118655)

Co-authored-by: J. Nick Koston <nick@koston.org>

rikroe committed:
Allow per-sensor unit conversion on BMW sensors (#110272)

* Update BMW sensors to use device_class

* Test adjustments

* Trigger CI

* Remove unneeded cast

* Set suggested_display_precision to 0

* Rebase for climate_status

* Change charging_status to ENUM device class

* Add test for Enum translations

* Pin Enum sensor values

* Use snapshot_platform helper

* Remove translation tests

* Formatting

* Remove comment

* Use const.STATE_UNKOWN

* Fix typo

* Update strings

* Loop through Enum sensors

* Revert enum sensor changes

---------

Co-authored-by: Richard <rikroe@users.noreply.github.com>

joostlek committed:
Initial cleanup for Aladdin connect (#118777)

joostlek committed:
Use model from Aladdin Connect lib (#118778)

* Use model from Aladdin Connect lib

* Fix

joostlek committed:
Add coordinator to Aladdin Connect (#118781)

joostlek committed:
Bump airgradient to 0.4.3 (#118776)

joostlek committed:
Re-enable sensor platform for Aladdin Connect (#118782)

joostlek committed:
Move Aladdin stale device removal to init module (#118784)

frenck committed:
Bump version to 2024.6.0b7

joostlek committed:
Set unique id in aladdin connect config flow (#118798)

balloob committed:
Include script description in LLM exposed entities (#118749)

* Include script description in LLM exposed entities

* Fix race in test

* Fix type

* Expose script

* Remove fields

agners committed:
Bump Python Matter Server library to 6.1.0 (#118806)

bramkragten committed:
Update frontend to 20240604.0 (#118811)

bieniu committed:
Check if Shelly `entry.runtime_data` is available (#118805)

* Check if runtime_data is available

* Add tests

* Use `is` operator

---------

Co-authored-by: Maciej Bieniek <478555+bieniu@users.noreply.github.com>

bdraco committed:
Ensure name of task is logged for unhandled loop exceptions (#118822)

jpbede committed:
Initialize the Sentry SDK within an import executor job to not block event loop (#118830)

tsvi committed:
Fix updating options in Jewish Calendar (#118643)

arturyak committed:
Add missing FAN_ONLY mode to ccm15 (#118804)

boswelja committed:
Fix calculation of Starlink sleep end setting (#115507)

Co-authored-by: J. Nick Koston <nick@koston.org>

frenck committed:
Bump version to 2024.6.0b8

frenck committed:
Fix capitalization of protocols in Reolink option flow (#118839)

starkillerOG committed:
Conserve Reolink battery by not waking the camera on each update (#118773)

* update to new cmd_list type

* Wake battery cams each 1 hour

* fix styling

* fix epoch

* fix timezone

* force full update when using generic update service

* improve comment

* Use time.time() instead of datetime

* fix import order

balloob committed:
Fix the radio browser doing I/O in the event loop (#118842)

Commit:
Bump python-roborock to 2.2.3 (#118853)

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

cdce8p committed:
Fix TypeAliasType not callable in senz (#118872)

bramkragten committed:
Update frontend to 20240605.0 (#118875)

PeteRager committed:
Fix Hydrawise sensor availability (#118669)

Co-authored-by: Robert Resch <robert@resch.dev>

jbouwh committed:
Improve repair issue when notify service is still being used (#118855)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
Bump version to 2024.6.0b9

edenhaus committed:
Replace slave by meter in v2c (#118893)

synesthesiam committed:
Bump intents to 2024.6.5 (#118890)

emontnemery committed:
Improve WS command validate_config (#118864)

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>
Co-authored-by: Robert Resch <robert@resch.dev>

frenck committed:
Bump version to 2024.6.0

Kane610 reviewed (approved):
LGTM 🎉🎉

jpbede reviewed (approved):
🎉🥳
